app/models/cinstance.rb

```diff
@@ -111,17 +111,20 @@ def validate_plan_is_unique?
   validate :same_service, on: :update, if: :plan_id_changed?
 
   # letter, number, underscore (_), hyphen-minus (-), dot (.), base64 format
-  # In base64 encoding, the character set is [A-Z,a-z,0-9,and + /], if rest length is less than 4, fill of '=' character.
-  # ^([A-Za-z0-9+/]{4})* means the String start with 0 time or more base64 group.
-  # ([A-Za-z0-9+/]{4}|[A-Za-z0-9+/]{3}=|[A-Za-z0-9+/]{2}==) means the String must end of 3 forms in [A-Za-z0-9+/]{4} or [A-Za-z0-9+/]{3}= or [A-Za-z0-9+/]{2}==
+  # In base64 encoding, the character set is [A-Z,a-z,0-9,and +], if rest length is less than 4, fill of '=' character.
+  # ^([A-Za-z0-9+]{4})* means the String start with 0 time or more base64 group.
+  # ([A-Za-z0-9+]{4}|[A-Za-z0-9+]{3}=|[A-Za-z0-9+]{2}==) means the String must end of 3 forms in [A-Za-z0-9+]{4} or [A-Za-z0-9+]{3}= or [A-Za-z0-9+]{2}==
   # matches also the non 64B case with (\A[\w\-\.]+\Z)
-  USER_KEY_FORMAT = /(([\w\-\.]+)|([A-Za-z0-9+\/]{4})*([A-Za-z0-9+\/]{4}|[A-Za-z0-9+\/]{3}=|[A-Za-z0-9+\/]{2}==))/
-
-  validates :application_id, format: { with: /\A[\x20-\x7E]+\Z/ }, length: { in: 4..255 }
+  # NOTE: base64 format also accepts forward slash (/), however we don't allow it because of the restriction on the backend
+  USER_KEY_FORMAT = /(([\w\-.]+)|([A-Za-z0-9+]{4})*([A-Za-z0-9+]{4}|[A-Za-z0-9+]{3}=|[A-Za-z0-9+]{2}==))/.freeze
 
+  # The following characters are accepted:
+  # A-Z a-z 0-9 ! " # $ % & ' ( ) * + , - . : ; < = > ? @ [ \ ] ^ _ ` { | } ~
+  # Spaces and / are not allowed
+  validates :application_id, format: { with: /\A[\x21-\x2E\x30-\x7E]+\Z/ }, length: { in: 4..255 }
```

Comment:
This could prevent updating existing applications as well. I've been looking for any `touch` among `Cininstance` associations that could try to update the application as a result of some other action. The only thing I found is this:

https://github.com/3scale/porta/blob/18c529f2f20289ba34225685a292454eb6c3cfcd/app/models/application_key.rb#L78

I'm not sure what that `proxy_associtation.owner` means, could it be the `Cinsntance`? If that's the case Oauth applications won't be able to regenerate keys on if they have spaces or slashes in their ids.

Reply:
Good catch, I'll check that.